### ISSUE TITLE
[MacOS][SDL2] Blurred rendering with SDL2 on retina displays

### ISSUE THREAD

**Issue by grauw** [openMSX/openMSX#1130]
Macs with retina displays have twice the pixel density normal displays have.

Compare SDL1:
<img width="752" alt="schermafbeelding 2018-12-01 om 12 23 14" src="https://user-images.githubusercontent.com/772052/49327886-5fa6be80-f567-11e8-97ee-6a81e4c0fe72.png">
With SDL2:
<img width="752" alt="schermafbeelding 2018-12-01 om 12 53 29" src="https://user-images.githubusercontent.com/772052/49327929-291d7380-f568-11e8-830b-0a692e7b5818.png">

As you can see in SDL2 everything looks much blurrier. This is probably due to SDL1 upscaling with nearest neighbour interpolation, whereas in SDL2 it is interpolated bilinearly.

The ideal solution would be to double the internal rendering resolution. In other words kind of an internal 2x multiplier on the scale_factor setting, while not affecting the window size. The aspect ratio correction and scanlines will also look nicer that way, really sharp.

**Comment by MBilderbeek** [comment]
@grauw Is this still the same with the latest SDL2 and XCode?

**Comment by grauw** [comment]
Yes, I just made a new build on ac4f9d914 and it’s still there. Although I’ve kinda gotten used to it, it does still look significantly better when I switch to the SDL renderer.

**Comment by mbabuskov** [comment]
I know this isn't the SDL forum, but I'm a game developer I can confirm the same problem with my game. When I grab the screenshot using SDL_RenderReadPixels() into a file, that screenshot has sharp graphics. But the graphics displayed on the screen is blurry. It wasn't always like this, when I used SDL 2.0.5, it rendered clear.

I tried changing the renderer using SDL_HINT_RENDER_DRIVER hint. It looks like "metal" is the default in newer SDL2 versions, and that one is the most blurry. I tried "opengles", which is a little bit better but still has some blur, and "opengl" which seems to render like it used before.

So, this isn't really SDL1 vs SDL2 issue, but it changed somewhere during SDL2 development. Most likely when they changed the default renderer. 

Hope this helps.

**Comment by MBilderbeek** [comment]
@grauw  is this still an issue?

**Comment by grauw** [comment]
It still looks the same as in the 2nd screenshot, yes.
